### PR TITLE
astyle: update to 3.1

### PR DESCRIPTION
**Pull request by reneeotten**
#### Description
- update astyle to latest version
- use compilers / cxx11 portgroups ; removed universal portgroup

Closes: https://trac.macports.org/ticket/53291

It would be good if someone checked the correctness of the portgroup usage (and perhaps whether it builds correctly on older systems). @l2dy since you opened the original ticket, perhaps you want to take a peak?

###### Tested on

macOS 10.13.6 17G65
Xcode 10.0 10A255

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: devel/astyle/files/patch-Makefile.diff [hunk 1]
```diff
@@ -1,58 +1,71 @@
...
  ipath=$(prefix)/bin
- CBASEFLAGS = -W -Wall -fno-rtti -fno-exceptions
--UNIVFLAGS = -arch i386 -arch x86_64
- JAVAINCS   = -I/System/Library/Frameworks/JavaVM.framework/Headers
+ CBASEFLAGS = -W -Wall -std=c++11 -stdlib=libc++ -fno-rtti -fno-exceptions  \
+ -mmacosx-version-min=10.9
+-LDBASEFLAGS = 
++LDBASEFLAGS =
+ JAVAINCS  = -I/Library/Java/JavaVirtualMachines/jdk1.8.0_45.jdk/Contents/Home/include \
+ -I/Library/Java/JavaVirtualMachines/jdk1.8.0_45.jdk/Contents/Home/include/darwin \
```
Comment (l2dy) [inline devel/astyle/files/patch-Makefile.diff]: Why hardcode `jdk1.8.0_45.jdk`? I don't think most users would have this particular version installed.

**Comment by reneeotten** [comment]
@l2dy good point - it was actually hardcoded in the Makefile, but I missed that. I have changed it to 
```JAVAINCS  = -I$(JAVA_HOME)/include -I$(JAVA_HOME)/include/darwin```, which will respect the value for ```JAVA_HOME``` set by the java portgroup.

**Comment by pmetzger** [comment]
Anything else needed before we merge this?

**Comment by pmetzger** [comment]
@reneeotten let me know when you feel comfortable having this merged.

**Comment by l2dy** [comment]
Is JDK 1.8 required? i.e. Does JDK 11 builds this port? If not, please add `java.version  1.8` to exclude newer JDKs.

**Comment by reneeotten** [comment]
trying to find this information I noticed that the license has changed to ```MIT``` - I've updated that in the PR. 

Regarding the JDK - to be honest I have no idea. The ```java``` part is in a variant which I don't use and from the [website](http://astyle.sourceforge.net/develop/jni.html) it doesn't become clear to me... I don't have any other JDK installed than ```jdk1.8.0_111``` so can't tell if it works with anything newer. I am fine with adding ```java.version 1.8``` if you think that's a good idea for now.

**Comment by l2dy** [comment]
> Regarding the JDK - to be honest I have no idea. The `java` part is in a variant which I don't use and from the [website](http://astyle.sourceforge.net/develop/jni.html) it doesn't become clear to me... I don't have any other JDK installed than `jdk1.8.0_111` so can't tell if it works with anything newer. I am fine with adding `java.version 1.8` if you think that's a good idea for now.

Then I'd suggest not adding it for now.

**Comment by reneeotten** [comment]
okay. Also, is anyone concerned about me removing the ```universal``` portgroup? To be honest, I don't fully understand its purpose/implications... other than that, it "works for me".

**Comment by pmetzger** [comment]
It doesn't bug me but I don't fully understand the implications either.

**Comment by pmetzger** [comment]
Merged. Thanks, @reneeotten!

**Comment by ryandesign** [comment]
Why was the compilers portgroup added? It doesn't look like it's used.

The reason why one would add the *m*universal portgroup is to add a universal variant for build systems which do not understand how to build for multiple architectures simultaneously. muniversal builds for each arch separately and *m*erges them together. Perhaps there was a change in the build system since it was added, because at the moment the port does not use the correct `-arch` flags regardless of whether the muniversal portgroup is used (nor the correct compiler). We should fix these problems.

We're currently using the project's custom Mac-specific Makefile, but there's also cmake support. Did you check whether using cmake works? If not, that's what I'll try, since I suspect it will fix these issues.

**Comment by ryandesign** [comment]
For portgroups that aren't described in the guide, read the portgroup tcl file; often they have documentation at the top or embedded into the code as comments.

The purpose of the compilers portgroup is to offer the user the choice of what compiler to use, among the options defined by the portfile author using the `compilers.setup` procedure. This is a very esoteric need; there are very few ports where we would actually want to offer that choice. Some extremely performance-sensitive ports do so, as do some ports that require Fortran, since only the gcc ports provide a Fortran compiler and if the user already has one installed we don't want to force them to install another one.

For many reasons, we want all ports to use the compiler MacPorts tells them to. At present, astyle doesn't do that. This would prevent the compilers and cxx11 portgroups from functioning, for example. See [UsingTheRightCompiler](https://trac.macports.org/wiki/UsingTheRightCompiler).

It's also important for ports to use the `-arch` flags MacPorts tells them to. MacPorts wants to offer users the ability to install a port with the universal variant, or to build for a nondefault architecture (by setting `build_arch` in macports.conf). These are simply features MacPorts wants users to have available. They were added back when Intel Macs were introduced and are arguably less relevant today, but we still want to offer them. MacPorts automatically handles this for ports using autotools; the cmake and xcode portgroups automatically handle ports using those build systems. For custom build systems, such as the one the astyle port currently uses, the portfile maintainer is responsible for implementing this feature.

In some cases, MacPorts will automatically select the universal variant. For example, if you try to install a port that can only be built 32-bit, but it has dependencies on ports that can be built 64-bit, then MacPorts will automatically install those dependencies with the universal variant in order to provide them in both a 32-bit version (for that 32-bit port you wanted to install) and a 64-bit version (for 64-bit-capable ports that depend on it). Obviously this only works if the port has a functioning universal variant.

If you have other questions about how MacPorts works or how to do something, ask on the macports-dev mailing list. All MacPorts developers should be subscribed there. Far fewer developers will read all pull requests, so asking a question in a pull request will be much less likely to get you an answer.

I'll take a look at astyle's cmake build system to see if we can easily switch to that.

**Comment by reneeotten** [comment]
@ryandesign thank you for your patience and explanations and looking into it yourself! I will try and educate myself a bit more and/or ask questions on the mailinglist next time. I probably should just have left this one alone and stick to Python ports that don't need compiler stuff for now, as this was obviously a bit too far out of my comfort zone ;)

**Comment by ryandesign** [comment]
> I started with the CMake build system a while back, but run into trouble at some point (and can't remember the exact issue anymore).

I'm not surprised. Their CMakeLists.txt was terrible. See #3005.

**Pull request merged**